control/nlsys.py

```diff
@@ -1308,9 +1319,31 @@ def nlsys(
     >>> timepts = np.linspace(0, 10)
     >>> response = ct.input_output_response(
     ...     kincar, timepts, [10, 0.05 * np.sin(timepts)])
+
     """
-    return NonlinearIOSystem(
-        updfcn, outfcn, inputs=inputs, outputs=outputs, states=states, **kwargs)
+    from .statesp import StateSpace
+    from .iosys import _extended_system_name
+
+    if isinstance(updfcn, StateSpace):
+        sys_ss = updfcn
+        kwargs['inputs'] = kwargs.get('inputs', sys_ss.input_labels)
+        kwargs['outputs'] = kwargs.get('outputs', sys_ss.output_labels)
+        kwargs['states'] = kwargs.get('states', sys_ss.state_labels)
+        kwargs['name'] = kwargs.get('name', _extended_system_name(
+            sys_ss.name, prefix_suffix_name='converted'))
+
+        sys_nl = NonlinearIOSystem(
+            lambda t, x, u, params: sys.A @ x + sys.B @ u,
+            lambda t, x, u, params: sys.C @ x + sys.D @ u, **kwargs)
```

Comment:
```suggestion
            lambda t, x, u, params: sys_ss.A @ x + sys_ss.B @ u,
            lambda t, x, u, params: sys_ss.C @ x + sys_ss.D @ u, **kwargs)
```

Reply:
This is a bug and needs to be fixed before merging.

Reply:
Good catch on this.  In the latest update, I added a unit test that catches the bug (and fixed the bug).